airlock/views/workspace.py

```diff
@@ -43,7 +47,8 @@ def workspace_view(request, workspace_name: str, path: str = ""):
         template = "file_browser/contents.html"
         selected_only = True
 
-    tree = get_workspace_tree(workspace, path, selected_only)
+    with tracer.start_as_current_span("build_workspace_tree") as _span:
+        tree = get_workspace_tree(workspace, path, selected_only)
```

Comment:
What about moving this span collection *inside* get_workspace_tree? Then we get it every time it's called, rather than just this call site?

`start_as_span` can be used as a decorator on the function: https://opentelemetry.io/docs/languages/python/instrumentation/#creating-spans-with-decorators

Also, we can add some more useful information to this span too, like workspace and user

e.g. something like this

```
@tracer.start_as_current_span("get_workspace_tree")
def get_workspace_tree(workspace, path, selected_only):
    span = tracer.get_current_span()
    span.attributes.set("workspace", workspace)
    span.attributes.set("path", path)
    span.attributes.set("selected_only", selected_only)
    ...
```

Admittedly, collecting the attributes is a bit messy when using a decorator, so it may be nice to stick with the context manager inside the function. We could make a better decorator if we wanted to: https://betterprogramming.pub/using-decorators-to-instrument-python-code-with-opentelemetry-traces-d7f1c7d6f632

Reply:
Sure - I only added this one so that I could check the test setup worked; hadn't really thought about where it was best to go yet.  I'll maybe leave that for a separate PR.